### ISSUE TITLE
operater format problem

### ISSUE THREAD

**Issue by CodeEagle** [nicklockwood/SwiftFormat#388]
swiftformat(0.40.0) will format  this
```swift

func test() {
  var data: [UInt8] = []
  if data[4] == 0x4 || data[4] == 0x0 {
  }
}
```
to 
```swift
func test() {
    var data: [UInt8] = []
    if data0x4 || data == [4][4] == 0x0 {}
}
```

**Comment by CodeEagle** [comment]
also && orperation 
```swift
if ("\u{4E00}" <= value  && value <= "\u{9FA5}") {
          return true
}

//to
 if value, value <= "\u{9FA5}" >= "\u{4E00}" {
                return true
            }
```

**Comment by nicklockwood** [comment]
@codeeagle thanks for reporting. I believe the former of these two bugs is now fixed on the develop branch. I'm not sure about the latter - I'll check.

**Comment by nicklockwood** [comment]
@codeeagle in the meantime, I suggest disabling the yodaConditions rule, and possibly the andOperator rule for this file.

**Comment by CodeEagle** [comment]
hi @nicklockwood ， is `yodaConditions `. develop branch fixed it.

**Comment by nicklockwood** [comment]
@CodeEagle great. I've landed the fix in version 0.40.2.
